### ISSUE TITLE
Interested in supporting PEP 665?

### ISSUE THREAD

**Issue by brettcannon** [pypa/pipenv#4825]
I'm asking various packaging projects if they would support [PEP 665](https://discuss.python.org/t/pep-665-take-2-a-file-format-to-list-python-dependencies-for-reproducibility-of-an-application/11736). This is to gather quotes of support for the PEP to help drive its acceptance. If you think you may support the PEP if it were to be accepted, could you please let me know or leave a comment on the discussion?

Thanks!

**Comment by stewartmiles** [comment]
Hey @brettcannon [PEP 665](https://discuss.python.org/t/pep-665-take-2-a-file-format-to-list-python-dependencies-for-reproducibility-of-an-application/11736) is really interesting.

I recently built an extension to pipenv (currently private I'm afraid) that generates a lock file with graph to do pretty much what you propose in PEP 665. Though something that I ended up doing, that is potentially missing from PEP 665, is support for build-time (i.e pyproject.toml) requirements. Build-time requirements are important to include in the graph so that it's possible to lock those packages for editable installs. Furthermore, build time requirement need to be topologically sorted so that editable packages are installed after their build-time requirements are available when using pip's `PIP_NO_BUILD_ISOLATION=0` / `--no-build-isolation` option (required to get any sort of reasonable performance with editable packages with load of build time dependencies or editable packages that depend upon each other).

Any chance you would be open to including support for build time deps in PEP 665?
